src/providers/coinbase/coinbase.ts

```diff
@@ -761,4 +763,80 @@ export class CoinbaseProvider {
       });
     });
   }
+
+  public getAvailableAccounts(coin?, minFiatCurrency?: { amount; currency }) {
+    let coinbaseData = _.cloneDeep(this.coinbaseData);
+    let coinbaseAccounts = coinbaseData.accounts.filter(ac => {
+      const accountCoin = ac.balance.currency.toLowerCase();
+      if (minFiatCurrency) {
+        const availableBalanceFiat = this.rateProvider.toFiat(
+          this.currencyProvider.getPrecision(accountCoin).unitToSatoshi *
+            Number(ac.balance.amount),
+          minFiatCurrency.currency,
+          accountCoin
+        );
+        return (
+          availableBalanceFiat >=
+          Number(
+            minFiatCurrency && minFiatCurrency.amount
+              ? minFiatCurrency.amount
+              : null
+          )
+        );
+      } else if (coin) {
+        return accountCoin == coin;
+      } else return ac;
+    });
+    return coinbaseAccounts;
+  }
+
+  public payInvoice(
+    invoiceId: string,
+    currency: string,
+    twoFactorCode?: string
+  ): Promise<any> {
+    return this.doRefreshToken().then(_ => {
```

Comment:
If 2fa is required, this function will be called twice. I think that we could avoid that when 2fa was not null.